codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -468,6 +464,9 @@ public boolean write(int allowedBytes) {
                             size == bytesWritten && endOfStream, writePromise);
                 } while (size != bytesWritten && allowedBytes > bytesWritten);
                 size -= bytesWritten;
+                if (size == 0 && endOfStream) {
+                    lifecycleManager.closeLocalSide(stream, promise);
```

Comment:
@Scottmitch There ain't much duplication right now - two calls to `lifecycleManager.closeLocalStream()` instead of one. The `writeComplete` method would basically be a one liner.